p2p/pex_reactor.go

```diff
@@ -108,18 +118,55 @@ func (r *PEXReactor) AddPeer(p Peer) {
 	} else { // For inbound connections, the peer is its own source
 		addr, err := NewNetAddressString(p.NodeInfo().ListenAddr)
 		if err != nil {
-			// peer gave us a bad ListenAddr. TODO: punish
+			// peer gave us a bad ListenAddr
+			r.MarkPeer(p, false, 1, Bad)
 			r.Logger.Error("Error in AddPeer: invalid peer address", "addr", p.NodeInfo().ListenAddr, "err", err)
 			return
 		}
+		r.MarkPeer(p, true, 1, Correct)
 		r.book.AddAddress(addr, addr)
 	}
 }
 
+// MarkPeer implements Reactor.
+// It updates a peer's metric with good or bad events taking place within this reactor
+func (r *PEXReactor) MarkPeer(peer Peer, good bool, events, severity int) {
+	if r.Switch == nil {
+		return
+	}
+
+	tms := r.Switch.MetricStore()
+	if tms == nil {
+		return
+	}
+
+	// Modify the number of events based on severity
+	num := events
```

Comment:
is this the same for all reactors? maybe we should create a sep function then to avoid copy&paste?

Reply:
it's currently the same, but in the future, different reactors could handle it differently as it makes since to support their needs